File: svc-event/src/main/java/com/beancounter/event/service/Event.java
package com.beancounter.event.service;

import com.beancounter.common.event.CorporateEvent;
import com.beancounter.common.input.TrustedTrnEvent;
import com.beancounter.common.model.Portfolio;
import com.beancounter.common.model.Position;

public interface Event {
  TrustedTrnEvent calculate(Portfolio portfolio,
                            Position currentPosition,
                            CorporateEvent corporateEvent);
}

File: svc-event/src/main/java/com/beancounter/event/service/Event.kt
package com.beancounter.event.service

import com.beancounter.common.event.CorporateEvent
import com.beancounter.common.input.TrustedTrnEvent
import com.beancounter.common.model.Portfolio
import com.beancounter.common.model.Position

interface Event {
    fun calculate(portfolio: Portfolio,
                  currentPosition: Position,
                  corporateEvent: CorporateEvent): TrustedTrnEvent?
}